core/src/com/github/czyzby/tests/Main.java

```diff
@@ -234,6 +239,28 @@ public void process(final LmlParser parser, final LmlTag tag, final BlinkingLabe
         };
     }
 
+    //Provides tag for custom CodeTextArea
+    private static LmlTagProvider getCodeTextAreaProvider() {
+        return new LmlTagProvider() {
+            @Override
+            public LmlTag create(final LmlParser parser, final LmlTag parentTag, final String rawTagData) {
+                return getCodeTextAreaTag(parser, parentTag, rawTagData);
+            }
+        };
+    }
+
+    //Custom TextArea supporting embedding in scroll pane, used to display code of current LML example
+    private static LmlTag getCodeTextAreaTag(final LmlParser parser, final LmlTag parentTag, final String rawTagData) {
+        return new VisTextAreaLmlTag(parser, parentTag, rawTagData) {
+            @Override
+            protected VisTextField getNewInstanceOfTextField(TextLmlActorBuilder textBuilder) {
+                final CodeTextArea textArea = new CodeTextArea(textBuilder.getText(), textBuilder.getStyleName());
+                LmlUtilities.getLmlUserObject(textArea).setData(Boolean.TRUE); // Setting as multiline by default.
+                return textArea;
+            }
+        };
+    }
+
```

Comment:
I'd move the tag provider into a nested class in CodeTextArea; this is mostly for internal use, so no need to keep this in the main class.